File: app/src/main/java/me/codeminions/mvpproject/mvp/app/BaseActivity.java
package me.codeminions.mvpproject.mvp.app;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    @LayoutRes
    protected abstract int getLayoutRes();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutRes());

        initWidget();
        initData();
    }

    protected void initWidget(){

    }

    protected void initData(){

    }
}

File: common/src/main/java/me/codeminions/common/app/BaseActivity.kt
package me.codeminions.common.app

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity: AppCompatActivity() {

    @LayoutRes
    abstract fun setLayoutId():Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(setLayoutId())

        initWidget()
        intiData()
    }

    open fun initWidget(){

    }

    open fun intiData(){

    }
}